modules/ximgproc/src/structured_edge_detection.cpp

```diff
@@ -810,6 +830,171 @@ class StructuredEdgeDetectionImpl : public StructuredEdgeDetection
+
+    /**
+     * Load model from a file.
+     * \param filename [in] Model file name.
+     * \param rf       [out] Model where parameters are saved.
+     */
+    void loadModelFromBinaryFile(const String &filename, struct RandomForest &rf)
+    {
+        FILE *f = fopen(filename.c_str(), "rb");
+        CV_Assert(f);
+
+        int help;
+        size_t ret;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.stride = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.shrinkNumber = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.patchSize = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.patchInnerSize = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.numberOfGradientOrientations = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.gradientSmoothingRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.regFeatureSmoothingRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.ssFeatureSmoothingRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.gradientNormalizationRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.selfsimilarityGridSize = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.numberOfTrees = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.numberOfTreesToEvaluate = help;
+
+        rf.options.numberOfOutputChannels = 2*(rf.options.numberOfGradientOrientations+1)+3;
+
+        size_t len;
+        uint64_t len_;
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.childs.resize(len);
+        ret = fread(rf.childs.data(), sizeof(int), len, f);
+
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.featureIds.resize(len);
+        ret = fread(rf.featureIds.data(), sizeof(int), len, f);
+
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.thresholds.resize(len);
+        ret = fread(rf.thresholds.data(), sizeof(float), len, f);
+
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.edgeBoundaries.resize(len);
+        ret = fread(rf.edgeBoundaries.data(), sizeof(int), len, f);
+
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.edgeBins.resize(len);
+        ret = fread(rf.edgeBins.data(), sizeof(int), len, f);
+        CV_Assert(ret == len);
+
+        rf.numberOfTreeNodes = int( rf.childs.size() ) / rf.options.numberOfTrees;
+
+        fclose(f);
+    }
+
+    /**
+     * Save model to a file.
+     * \param filename  [in] Model file name.
+     * \param rf        [in] Model parameters.
+     */
+    void saveModelToBinaryFile(const String &filename, const struct RandomForest &rf)
```

Comment:
Yes, you are right. I used this method initially to convert the the existing text format model to binary.
Now a Matlab script is used to do this job. This method is removed in the new commit.